src/SingleSelect.elm

```diff
@@ -345,30 +322,68 @@ removeSelectedFromOptions selectedOption options =
         |> Maybe.withDefault options
 
 
-filterAndIndexOptions : Model msg a -> List a -> List ( Int, a )
-filterAndIndexOptions model unfilteredOptions =
-    if model.searchText == "" then
-        removeSelectedFromOptions model.selected unfilteredOptions
+filterAndIndexOptions : { options : List a, selectedOption : Maybe a, searchFn : String -> List a -> List a, searchText : String } -> List ( Int, a )
+filterAndIndexOptions { options, selectedOption, searchFn, searchText } =
+    if searchText == "" then
+        removeSelectedFromOptions selectedOption options
             |> List.indexedMap Tuple.pair
 
     else
-        model.settings.searchFn model.searchText unfilteredOptions
-            |> removeSelectedFromOptions model.selected
+        searchFn searchText options
+            |> removeSelectedFromOptions selectedOption
             |> List.indexedMap Tuple.pair
 
 
-{-| The smart select view for selecting one option at a time with local data. It expects the following arguments (in order):
+{-| The smart select view for selecting one option at a time with local data.
 
-  - a boolean indicating if the select is disabled or not
-  - a list of the items from which to select
-  - the smart select instance
+  - `options` takes a list of the data being selected from.
+  - `optionLabelFn` takes a function that expects an instance of the data being selected from and returns a string naming/labeling the instance, i.e. if it is a "Product" being selected, the label may be "Garden Hose".
 
 -}
-view : Bool -> List a -> SmartSelect msg a -> Html msg
-view isDisabled allOptions (SmartSelect model) =
+view : { options : List a, optionLabelFn : a -> String } -> SmartSelect msg a -> Html msg
+view { options, optionLabelFn } smartSelect =
+    let
+        config =
+            { isDisabled = False
+            , options = options
+            , optionType = "Option"
```

Comment:
Basically two things on this..
1. I don't like this default value. A empty string feels better. Just like a usual HTML select input - it either has a specific input or is blank. And I know that this is used for things like the search "Search for a Option.." etc. - so point 2

2. Why don't we keep the smartselect texts more generic ("Search ...", "No entries found." ...) and use the "optionType" just to be displayed when the smartselect is closed and nothing is selected? Firstly this avoids that non-generic names can end up grammatically wrong ("No mediums are available"). Secondly the "optionType" can be used more flexible

Reply:
Alright... before I go off and do this, I'd like to get a consensus on what I've been thinking. In order to facilitate i18n and be more flexible, I think it makes sense for users to pass in their own search prompts, etc. Let me know what you think @Tranquility @mschindlerMM 

so there would be the following for local selects:
```elm
selectTitle : String -- replaces optionType, for multi, would show up when select is closed and empty
searchPrompt : String
noResultsForMsg : String -> String
noOptionsMsg : String -- primarily for multi in the event all options have been selected, but could occur for single in the event an empty options list has been provided
```

and for the remote selects:
```elm
selectTitle : String -- replaces optionType, for multi, would show up when select is closed and empty
searchPrompt : Int -> String
queryErrorMsg : String
noResultsForMsg : String -> String
noOptionsMsg : String -- primarily for multi in the event all options have been selected, but could occur for single in the event an empty options list has been provided
```

Of course, defaults would be set for all of these in the basic view, and customizable in custom view

Reply:
Seems fine to me but maybe it's still better to use empty strings by default because there are no i18n issues with those :)